### PR TITLE
Update Flutter to use the latest dartdoc

### PR DESCRIPTION
**Pull request by abarth**
Now that dart-lang/dartdoc#1236 is fixed.

### CODE REVIEW

**Comment by devoncarew** [comment]
lgtm!

**Comment by sethladd** [comment]
thanks!

File: dev/bots/docs.sh [hunk 1]
```diff
@@ -2,15 +2,35 @@
...
-# so are we pinned to this old version until that bug is fixed.
-pub global activate dartdoc 0.9.7+1
+pub global activate dartdoc 0.9.9
 
 # This script generates a unified doc set, and creates
 # a custom index.html, placing everything into dev/docs/doc
 (cd dev/tools; pub get)
 FLUTTER_ROOT=$PWD dart dev/tools/dartdoc.dart
 
+# Smoke test the docs to make sure we have all the major kinds of things.
```
Comment (Hixie) [inline dev/bots/docs.sh]: didn't we have this test already somewhere?

Comment (abarth) [inline dev/bots/docs.sh]: Yes.  Sorry I missed it.  Will merge.

**Comment by abarth** [comment]
Fix in https://github.com/flutter/flutter/pull/7443

**Pull request merged**